unraid_mcp/tools/unraid.py

```
@@ -0,0 +1,1891 @@
+
+    if subaction == "disk_details" and not disk_id:
+        raise ToolError("disk_id is required for disk/disk_details")
+
+    if subaction == "logs":
+        if tail_lines < 1 or tail_lines > _MAX_TAIL_LINES:
+            raise ToolError(f"tail_lines must be between 1 and {_MAX_TAIL_LINES}, got {tail_lines}")
+        if not log_path:
+            raise ToolError("log_path is required for disk/logs")
+        normalized = os.path.realpath(log_path)  # noqa: ASYNC240
+        if not any(normalized.startswith(p) for p in _ALLOWED_LOG_PREFIXES):
+            raise ToolError(f"log_path must start with one of: {', '.join(_ALLOWED_LOG_PREFIXES)}")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)